go/types/set_iterator_test.go

```diff
@@ -0,0 +1,174 @@
+// Copyright 2016 Attic Labs, Inc. All rights reserved.
+// Licensed under the Apache License, version 2.0:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package types
+
+import (
+	"testing"
+
+	"github.com/attic-labs/testify/assert"
+)
+
+func TestSetIterator(t *testing.T) {
+	assert := assert.New(t)
+
+	numbers := append(generateNumbersAsValues(6), Number(10), Number(15))
+	s := NewSet(numbers...)
+	i := s.Iterator()
+
+	vs := ValueSlice{}
+	for v := i.Next(); v != nil; v = i.Next() {
+		vs = append(vs, v)
+	}
```

Comment:
Pattern is repeated a lot, how about a method like `iterToSlice(SetIterator) ValueSlice`? then just `vs := iterToSlice(i)` here.